sklearn/kernel_approximation.py

```diff
@@ -358,6 +329,156 @@ def _transform_sparse(self, X):
         return sp.hstack(X_new)
 
 
+class AdditiveChi2Sampler(BaseAdditiveHomogenousKernelSampler):
+    """Approximate feature map for additive chi2 kernel.
+
+    Uses sampling the fourier transform of the kernel characteristic
+    at regular intervals.
+
+    Since the kernel that is to be approximated is additive, the components of
+    the input vectors can be treated separately.  Each entry in the original
+    space is transformed into 2*sample_steps+1 features, where sample_steps is
+    a parameter of the method. Typical values of sample_steps include 1, 2 and
+    3.
+
+    Optimal choices for the sampling interval for certain data ranges can be
+    computed (see the reference). The default values should be reasonable.
+
+    Read more in the :ref:`User Guide <additive_chi_kernel_approx>`.
+
+    Parameters
+    ----------
+    sample_steps : int, optional
+        Gives the number of (complex) sampling points.
+    sample_interval : float, optional
+        Sampling interval. Must be specified when sample_steps not in {1,2,3}.
+
+    Notes
+    -----
+    This estimator approximates a slightly different version of the additive
+    chi squared kernel then ``metric.additive_chi2`` computes.
+
+    See also
+    --------
+    SkewedChi2Sampler : A Fourier-approximation to a non-additive variant of
+        the chi squared kernel.
+
+    sklearn.metrics.pairwise.chi2_kernel : The exact chi squared kernel.
+
+    sklearn.metrics.pairwise.additive_chi2_kernel : The exact additive chi
+        squared kernel.
+
+    References
+    ----------
+    See `"Efficient additive kernels via explicit feature maps"
+    <http://www.robots.ox.ac.uk/~vedaldi/assets/pubs/vedaldi11efficient.pdf>`_
+    A. Vedaldi and A. Zisserman, Pattern Analysis and Machine Intelligence,
+    2011
+    """
+
+    # See reference, figure 2 c)
+    preset_sample_intervals = {1: 0.8, 2: 0.5, 3: 0.4}
+
+    def _spectrum(self, omega):
+        # Spectrum function for chi2 kernel
+        return 1. / np.cosh(np.pi * omega)
+
+
+class IntersectionSampler(BaseAdditiveHomogenousKernelSampler):
+    """Approximate feature map for intersection kernel.
+
+    Uses sampling the fourier transform of the kernel characteristic
+    at regular intervals.
+
+    Since the kernel that is to be approximated is additive, the components of
+    the input vectors can be treated separately.  Each entry in the original
+    space is transformed into 2*sample_steps+1 features, where sample_steps is
```

Comment:
Add \`\` between `2 * sample_steps +1` and `sample_steps`.
Can you check that everywhere ?